src/ocaml_dependencies.ml

```diff
@@ -195,66 +195,30 @@ module Make (I : INPUT) = struct
     let dependencies_of x =
       try SMap.find x !*dependencies with Not_found -> Resources.empty in
 
-    let refine_cycle files starting_file =
-      (* We are looking for a cycle starting from [fn], included in
-         [files]; we'll simply use a DFS which builds a path until it
-         finds a circularity.
-
-         Note that if there is at least one cycle going through [fn],
-         calling [dfs path fn] will return it no matter what [path] is
-         (it may just not be the shortest possible cycle). This means
-         that if [dfs path fn] returns [None], [fn] is a dead-end that
-         should never be explored again.
-       *)
-      let dead_ends = ref Resources.empty in
-      let rec dfs path fn =
-        let through_dep f = function
-          | Some _ as cycle -> cycle
-          | None ->
-             if List.mem f path
-             then (* we have found a cycle *)
-               Some (List.rev path)
-             else if not (Resources.mem f files)
-             then
-               (* the neighbor is not in the set of paths known to have a cycle *)
-               None
-             else
-               (* look for cycles going through this neighbor *)
-               dfs (f :: path) f
-        in
-        if Resources.mem fn !dead_ends then None
-        else match Resources.fold through_dep (dependencies_of fn) None with
-          | Some _ as cycle -> cycle
-          | None -> dead_ends := Resources.add fn !dead_ends; None
-      in
-      match dfs [] starting_file with
-        | None -> Resources.elements files
-        | Some cycle -> cycle
-    in
-
     let needed_in_order = ref [] in
     let needed = ref Resources.empty in
-    let seen = ref Resources.empty in
-    let rec aux fn =
+    let rec aux on_the_go fn =
       if sys_file_exists fn && not (Resources.mem fn !needed) then begin
```

Comment:
I feel a bit lost re-reading this code after a while. Why is it testing `sys_file_exists fn` here, and only adding the files of `(dependencies_of ...)` if they do *not* exists below?